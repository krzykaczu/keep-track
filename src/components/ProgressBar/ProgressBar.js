import React from 'react';
import styled from 'styled-components';
import Range from '../Range';
import PropTypes from 'prop-types';

const StyledProgressBar = styled.div`
  width: 90%;
  height: 20px;
  border-radius: 10px;
  border: 1px solid #666;
  box-shadow: 0 0 1px 2px rgba(0, 0, 0, 0.1);
`;
const ProgressBar = ({ range, status }) => {
  return (
    <StyledProgressBar>
      <Range range={range} status={status} />
    </StyledProgressBar>
  );
};

ProgressBar.propTypes = {
  range: PropTypes.number.isRequired,
  status: PropTypes.string,
};

export default ProgressBar;
