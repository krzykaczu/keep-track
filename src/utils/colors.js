export const colors = {
  alert: {
    gradientStart: '#ff6699',
    gradientEnd: '#b30000',
  },
  warning: {
    gradientStart: '#ffeb99',
    gradientEnd: '#ffcc00',
  },
  normal: {
    gradientStart: '#9fdfbf',
    gradientEnd: '#008000',
  },
};
