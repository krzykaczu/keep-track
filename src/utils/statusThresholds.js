export const statusThresholds = {
  setAlert: 7,
  setWarning: 14,
};
