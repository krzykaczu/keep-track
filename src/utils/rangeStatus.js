import { colors } from './colors';

export const rangeStatus = {
  alert: `linear-gradient(-170deg, ${colors.alert.gradientStart} 0%, ${
    colors.alert.gradientEnd
  } 100%)`,
  warning: `linear-gradient(-170deg, ${colors.warning.gradientStart} 0%, ${
    colors.warning.gradientEnd
  } 100%)`,
  normal: `linear-gradient(-170deg, ${colors.normal.gradientStart} 0%, ${
    colors.normal.gradientEnd
  } 100%)`,
};
