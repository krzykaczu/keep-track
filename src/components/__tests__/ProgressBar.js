import React from 'react';
import ProgressBar from '../ProgressBar';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup } from 'react-testing-library';
import { colors } from '../../utils/colors';

afterEach(cleanup);

test('ProgressBar range props: 50 -> Range style width: 50%', () => {
  const { getByTestId, container } = render(<ProgressBar range={50} />);
  expect(container).toMatchSnapshot();
  expect(getByTestId('range')).toHaveStyle('width: 50%');
});

test('ProgressBar range props: 120 -> Range style width: 100%', () => {
  const { getByTestId } = render(<ProgressBar range={120} />);
  expect(getByTestId('range')).toHaveStyle('width: 100%');
});

test('ProgressBar range props: -10 -> Range style width: 0%', () => {
  const { getByTestId } = render(<ProgressBar range={-10} />);
  expect(getByTestId('range')).toHaveStyle('width: 0%');
});

test(`ProgressBar status props: alert -> Range style background: linear-gradient(-170deg, ${
  colors.alert.gradientStart
} 0%, ${colors.alert.gradientEnd} 100%)`, () => {
  const { getByTestId } = render(<ProgressBar range={50} status={'alert'} />);
  expect(getByTestId('range')).toHaveStyle(
    `background: linear-gradient(-170deg, ${colors.alert.gradientStart} 0%, ${
      colors.alert.gradientEnd
    } 100%)`
  );
});

test(`ProgressBar status props: warning -> Range style background: linear-gradient(-170deg, ${
  colors.warning.gradientStart
} 0%, ${colors.warning.gradientEnd} 100%)`, () => {
  const { getByTestId } = render(<ProgressBar range={50} status={'warning'} />);
  expect(getByTestId('range')).toHaveStyle(
    `background: linear-gradient(-170deg, ${colors.warning.gradientStart} 0%, ${
      colors.warning.gradientEnd
    } 100%)`
  );
});

test(`ProgressBar with no status props -> Range style background: linear-gradient(-170deg, ${
  colors.normal.gradientStart
} 0%, ${colors.normal.gradientEnd} 100%)`, () => {
  const { getByTestId } = render(<ProgressBar range={50} />);
  expect(getByTestId('range')).toHaveStyle(
    `background: linear-gradient(-170deg, ${colors.normal.gradientStart} 0%, ${
      colors.normal.gradientEnd
    } 100%)`
  );
});
