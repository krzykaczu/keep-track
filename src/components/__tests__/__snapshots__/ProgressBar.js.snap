// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar range props: 50 -> Range style width: 50% 1`] = `
.c1 {
  background: linear-gradient(-170deg,#9fdfbf 0%,#008000 100%);
  width: 50%;
  height: 100%;
  border-radius: inherit;
}

.c0 {
  width: 90%;
  height: 20px;
  border-radius: 10px;
  border: 1px solid #666;
  box-shadow: 0 0 1px 2px rgba(0,0,0,0.1);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="range"
    />
  </div>
</div>
`;
